pkg/api/types/containerd_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// Forked from https://github.com/containerd/containerd/blob/main/pkg/cri/store/container/metadata.go
+// Copyright The containerd Authors.
+// Licensed under the Apache License, Version 2.0
+// NOTE: we just want to get the key information of metadata( such as logpath), not all the metadata.
+
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// NOTE(random-liu):
+// 1) Metadata is immutable after created.
+// 2) Metadata is checkpointed as containerd container label.
+
+// metadataVersion is current version of container metadata.
+const metadataVersion = "v1" // nolint
+
+// versionedMetadata is the internal versioned container metadata.
+// nolint
+type versionedMetadata struct {
+	// Version indicates the version of the versioned container metadata.
+	Version string
+	// Metadata's type is metadataInternal. If not there will be a recursive call in MarshalJSON.
+	Metadata metadataInternal
+}
+
+// metadataInternal is for internal use.
+type metadataInternal Metadata
+
+// Metadata is the unversioned container metadata.
+type Metadata struct {
```

Comment:
```suggestion
type CRIContainerMetadata struct {
```

Reply:
Or split this to a new package like `pkg/api/types/critypes`

Reply:
create a new package `"github.com/containerd/nerdctl/pkg/api/types/cri"`